File: programers_Lv2/프로세스.py
def solution(priorities, location):
    answer = 0
    new_p = []
    for i, v in enumerate(priorities):
        new_p.append((v, i))
    value = new_p[location]

    while True:
        idx = priorities.index(max(priorities))
        temp = new_p[idx]
        priorities = priorities[idx + 1 :] + priorities[:idx]
        new_p = new_p[idx + 1 :] + new_p[:idx]

        answer += 1
        if value == temp:
            break

    return answer


# print(solution([2, 1, 3, 2], 2))  # 1
print(solution([1, 1, 9, 1, 1, 1], 0))  # 5

File: programers_Lv2/프로세스.js
function solution(priorities, location) {
  var answer = 0;
  let newPriorities = [];
  priorities.forEach((v, i) => {
    newPriorities.push(`${v}-${i}`);
  });
  const value = newPriorities[location];

  while (1) {
    let idx = priorities.indexOf(Math.max(...priorities));
    let temp = newPriorities[idx];

    priorities = [...priorities.slice(idx + 1), ...priorities.slice(0, idx)];
    newPriorities = [
      ...newPriorities.slice(idx + 1),
      ...newPriorities.slice(0, idx),
    ];
    answer++;
    if (value === temp) break;
  }
  return answer;
}

console.log(solution([1, 1, 9, 1, 1, 1], 0)); // 5
